packages/moonstone/VideoPlayer/Video.js

```diff
@@ -0,0 +1,297 @@
+		 * * `duration` {Number} - Media's entire duration in seconds
+		 * * `error` {Boolean} - `true` if video playback has errored.
+		 * * `loading` {Boolean} - `true` if video playback is loading.
+		 * * `paused` {Boolean} - Playing vs paused state. `true` means the media is paused
+		 * * `playbackRate` {Number} - Current playback rate, as a number
+		 * * `proportionLoaded` {Number} - A value between `0` and `1`
+		 *	representing the proportion of the media that has loaded
+		 * * `proportionPlayed` {Number} - A value between `0` and `1` representing the
+		 *	proportion of the media that has already been shown
+		 *
+		 * Methods:
+		 * * `play()` - play video
+		 * * `pause()` - pause video
+		 * * `load()` - load video
+		 *
+		 * The [`source`]{@link moonstone/VideoPlayer.VideoBase.source} property is passed to
+		 * the video component as a child node.
+		 *
+		 * @type {Component}
+		 * @default 'video'
+		 * @public
+		 */
+		mediaComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.element]),
+
+		/**
+		 * Disable automatically playing the video after it has loaded
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		noAutoPlay: PropTypes.bool,
+
+		/**
+		 * The video source to be preloaded. Expects a `<source>` node.
+		 *
+		 * @type {String|Node}
+		 * @public
+		 */
+		preloadSource:  PropTypes.node,
+
+		/**
+		 * Called with a reference to the active [Media]{@link ui/Media.Media} component
+		 *
+		 * @type {Function}
+		 * @private
+		 */
+		setMedia: PropTypes.func,
+
+		/**
+		 * The video source to be played.
+		 *
+		 * Any children `<source>` elements will be sent directly to the `mediaComponent` as video
+		 * sources.
+		 *
+		 * See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source
+		 *
+		 * @type {Node}
+		 * @public
+		 */
+		source: PropTypes.node
+	}
+
+	static defaultProps = {
+		mediaComponent: 'video'
+	}
+
+	constructor (props) {
+		super(props);
+		this.isPlayerMounted = false;
+
+		this.loaded = {
+			video: false,
+			preload: false
+		};
+	}
+
+	componentDidMount () {
+		this.isPlayerMounted = true;
+		this.setMedia();
+	}
+
+	componentDidUpdate (prevProps) {
+		const {source, preloadSource} = this.props;
+		const {source: prevSource, preloadSource: prevPreloadSource} = prevProps;
+
+		const key = getKeyFromSource(source);
+		const prevKey = getKeyFromSource(prevSource);
+		const preloadKey = getKeyFromSource(preloadSource);
+		const prevPreloadKey = getKeyFromSource(prevPreloadSource);
+
+		// if there's source and it has changed and it's not the prior preloaded source
+		if (source && key !== prevKey) {
+			// if it wasn't the preload key
+			if (key !== prevPreloadKey) {
+				// flag it as unloaded
+				this.loaded.video = false;
+			} else {
+				// if the preload video is switched to the current, try to play it now because it
+				// won't fire an onLoadStart events since it has already loaded.
+				this.autoPlay();
```

Comment:
Good question! I'll have to try this out. This specific case address switching to a preloaded video which previously did not have `autoPlay` but now does. I don't know if it will auto play if the video was already loaded when `autoPlay` was not set on load.

Reply:
Okay. So it seems that adding `autoplay` after the fact _might_ work. It does in firefox but doesn't in chrome. See https://codepen.io/anon/pen/bMvzEb. The chrome failure may be unique to a newer version of chrome for UX reasons (to prevent sites from auto playing unwanted videos).

That all said, we can't call `load()` on a video that was already loaded without throwing away the preload. Barring other suggestions, I'm going to leave this as is for now.